core/sail/base/pom.xml

```diff
@@ -51,9 +51,49 @@
 			<groupId>org.slf4j</groupId>
 			<artifactId>slf4j-api</artifactId>
 		</dependency>
+
+		<dependency>
+			<groupId>junit</groupId>
+			<artifactId>junit</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.mockito</groupId>
+			<artifactId>mockito-core</artifactId>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>org.assertj</groupId>
+			<artifactId>assertj-core</artifactId>
+			<scope>test</scope>
+		</dependency>
+
+		<dependency>
+			<groupId>${project.groupId}</groupId>
+			<artifactId>rdf4j-repository-sail</artifactId>
+			<version>${project.version}</version>
+			<scope>test</scope>
+		</dependency>
+		<dependency>
+			<groupId>${project.groupId}</groupId>
+			<artifactId>rdf4j-repository-manager</artifactId>
+			<version>${project.version}</version>
+			<scope>test</scope>
+		</dependency>
 	</dependencies>
 	<build>
 		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-jar-plugin</artifactId>
+				<executions>
+					<execution>
+						<goals>
+							<goal>test-jar</goal>
+						</goals>
```

Comment:
Do we need a test jar?

Reply:
does this approach mean we need to build twice? Or that eclipse/IntelliJ won’t be able to build properly?

Reply:
We need a test jar because normally maven modules don't include the test code. There are tests which are _defined_ at a certain abstract level but which need to executed in concrete implemenations. To be able to inherit from a common set of  tests, we use these test jars. This basically replaces the testsuite module we had for this purpose, and make it easier to just execute these tests as part of the module itself, instead of in a separate compliance module.

And no, it doesn't mean we need to build twice, I've been careful (hopefully) not to introduce any cyclic dependencies. This is a pretty common maven approach, it builds fine in Eclipse and though I haven't tested in IntelliJ I'd be very surprised if it had trouble with it.